File: XamarinXamlPrev/Libs/Utility.fs
namespace XamarinXamlPrev
open System
open System.Collections.Generic
open System.Linq
open System.Text
open Xamarin.Forms
open Moonmile.XFormsProvider
open Moonmile.XForms
open System.IO
open System.Reflection

type ResourceLoader() =
    static member val Assembly:Assembly = null with get, set
    static member val Names= [||] with get, set

    static member GetObject(resourceName:string) =
        if ResourceLoader.Assembly = null then
            ResourceLoader.Assembly <- typeof<ResourceLoader>.GetTypeInfo().Assembly
            ResourceLoader.Names <- ResourceLoader.Assembly.GetManifestResourceNames()
        try
            let path = 
                ResourceLoader.Names 
                |> Seq.find( fun(x) -> x.EndsWith(resourceName, StringComparison.CurrentCultureIgnoreCase))
            ResourceLoader.Assembly.GetManifestResourceStream(path)
        with
        | _ -> null

    static member GetString(resourceName:string) =
        use st = ResourceLoader.GetObject(resourceName)
        if st <> null then
            use sr = new StreamReader(st)
            let xaml = sr.ReadToEnd()
            xaml
        else
            ""

File: CSharp/XamarinXamlPreviewCS/XamarinXamlPreviewCS/XamarinXamlPreviewCS/Libs/Utility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace XamarinXamlPreviewCS.Libs
{
    public static class ResourceLoader
    {
        static internal string[] Names { get; set; }
        static internal Assembly Assembly { get; set; }

        public static System.IO.Stream GetObject(string resourceName)
        {
            if (ResourceLoader.Assembly == null)
            {
                ResourceLoader.Assembly = typeof(ResourceLoader).GetTypeInfo().Assembly;
                ResourceLoader.Names = ResourceLoader.Assembly.GetManifestResourceNames();
            }
            try
            {
                string path = ResourceLoader.Names.First(x => x.EndsWith(resourceName, StringComparison.CurrentCultureIgnoreCase));
                return ResourceLoader.Assembly.GetManifestResourceStream(path);
            }
            catch
            {
                return null;
            }
        }

        public static string GetString(string resourceName)
        {
            using (var st = ResourceLoader.GetObject(resourceName))
            {
                if (st != null)
                {
                    using (var sr = new System.IO.StreamReader(st))
                    {
                        return sr.ReadToEnd();
                    }
                }
                else
                {
                    return "";
                }
            }
        }
    }
}
